services/guard/guard.go

```diff
@@ -0,0 +1,76 @@
+package guard
+
+import (
+	"crypto/rand"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/crypto/abstract"
+)
+
+// This file contains all the code to run a Guard service. The Guard receives takes a
+// request for the Guard password protection service and sends back the message received encrypted with the key for each service
+// in the server.
+
+// ServiceName is the name to refer to the Guard service.
+const ServiceName = "Guard"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newGuardService)
+	network.RegisterMessageType(&Request{})
+	network.RegisterMessageType(&Response{})
+}
+
+//This is the area where Z is generated for a server, it creates z, which is a bytestring of length n for each guard.
+
+// Guard is a structure that stores the guards secret key, z, to be used later in the process of hashing the clients requests
+type Guard struct {
+	*sda.ServiceProcessor
+	path string
+	z    []byte
```

Comment:
Should probably be capital so the structure can be stored later (not yet implemented).